website/addons/s3/serializer.py

```diff
@@ -0,0 +1,64 @@
+from website.util import web_url_for
+from website.addons.base.serializer import OAuthAddonSerializer
+from website.addons.s3 import utils
+
+class S3Serializer(OAuthAddonSerializer):
+    addon_short_name = 's3'
+
+    REQUIRED_URLS = []
+
+    @property
+    def addon_serialized_urls(self):
+        node = self.node_settings.owner
+        user_settings = self.node_settings.user_settings or self.user_settings
+
+        result = {
+            'accounts': node.api_url_for('s3_account_list'),
+            'create_bucket': node.api_url_for('create_bucket'),
+            'import_auth': node.api_url_for('s3_import_auth'),
+            'create': node.api_url_for('s3_add_user_account'),
+            'deauthorize': node.api_url_for('s3_deauthorize_node'),
+            'bucket_list': node.api_url_for('s3_folder_list'),
+            'set_bucket': node.api_url_for('s3_get_config'),
+            'files': node.web_url_for('collect_file_trees'),
+        }
+        if user_settings:
+            result['owner'] = web_url_for('profile_view_id',
+                uid=user_settings.owner._primary_key)
+        return result
+
+    def credentials_are_valid(self, user_settings):
+        if user_settings:
+            if len(user_settings.external_accounts) < 1:
+                return False
+            return any([utils.can_list(account.oauth_key, account.oauth_secret)
```

Comment:
Small but if we do get to a point of having multiple addons this should be a for loop or generator. This will evaluate the entire list, possibly making unnecessary request.